server/src/test/java/io/druid/client/CachingQueryRunnerTest.java

```diff
@@ -237,6 +306,9 @@ public boolean isUseCache()
     Assert.assertTrue(closable.isClosed());
     Assert.assertEquals(expectedRes.toString(), results.toString());
 
+    // wait for background caching finish
+    // wait at most 10 seconds to fail the test to avoid block overall tests
+    cacheMustBePutOnce.await(10, TimeUnit.SECONDS);
```

Comment:
I'm on my phone right now but doesn't await just return a boolean and not actually throw an error on timeout?